nvflare/app_common/job_launcher/client_process_launcher.py

```diff
@@ -21,23 +21,28 @@
 
 
 class ClientProcessJobLauncher(ProcessJobLauncher):
-    def get_command(self, launch_data, fl_ctx) -> (str, dict):
+    def get_command(self, job_meta, fl_ctx) -> (str, dict):
         new_env = os.environ.copy()
+        workspace_obj: Workspace = fl_ctx.get_prop(FLContextKey.WORKSPACE_OBJECT)
+        job_id = job_meta.get(JobConstants.JOB_ID)
+        app_custom_folder = workspace_obj.get_app_custom_dir(job_id)
+        if app_custom_folder != "":
+            add_custom_dir_to_path(app_custom_folder, new_env)
+
+        command = self.generate_client_command(job_meta, fl_ctx)
+        return command, new_env
+
+    def generate_client_command(self, job_meta, fl_ctx):
```

Comment:
if this is moved out as regular common method, then ClientDockerJobLauncher does not need to inherit ClientProcessJobLauncher?

Reply:
We could also move this to a "util" class, then share between ClientDockerJobLauncher and ClientProcessJobLauncher. But this only specifically apply to the job launcher purpose. Stick with this inherit definition.

Reply:
Moved the generate_client_command() to job_launcher_utils.py.